events/ecs_container_instance.go

```diff
@@ -0,0 +1,70 @@
+package events
+
+import (
+	"encoding/json"
+	"time"
+)
+
+type ECSContainerInstanceEvent struct {
+	Version    string                              `json:"version"`
+	ID         string                              `json:"id"`
+	DetailType string                              `json:"detail-type"`
+	Source     string                              `json:"source"`
+	Account    string                              `json:"account"`
+	Time       time.Time                           `json:"time"`
+	Region     string                              `json:"region"`
+	Resources  []string                            `json:"resources"`
+	Detail     ECSContainerInstanceEventDetailType `json:"detail"`
+}
+
+type ECSContainerInstanceEventDetailType struct {
+	AgentConnected       bool        `json:"agentConnected"`
+	Attributes           []Attribute `json:"attributes"`
+	ClusterARN           string      `json:"clusterArn"`
+	ContainerInstanceARN string      `json:"containerInstanceArn"`
+	EC2InstanceID        string      `json:"ec2InstanceId"`
+	RegisteredResources  []Resource  `json:"registeredResources"`
+	RemainingResources   []Resource  `json:"remainingResources"`
+	Status               string      `json:"status"`
+	Version              int         `json:"version"`
+	VersionInfo          VersionInfo `json:"versionInfo"`
+	UpdatedAt            time.Time   `json:"updatedAt"`
+}
+
+type Attribute struct {
+	Name string `json:"name"`
+}
+
+type Resource struct {
+	Name           string    `json:"name"`
+	Type           string    `json:"type"`
+	IntegerValue   int       `json:"integerValue,omitempty"`
+	StringSetValue []*string `json:"stringSetValue,omitempty"`
+}
+
+type VersionInfo struct {
+	AgentHash     string `json:"agentHash"`
+	AgentVersion  string `json:"agentVersion"`
+	DockerVersion string `json:"dockerVersion"`
+}
+
+// MarshalJSON implements cuustom marshaling to marshal the struct into JSON format while preserving an empty string slice in `StringSetValue` field.
+func (r Resource) MarshalJSON() ([]byte, error) {
```

Comment:
At first glance I thought this was overkill for the sake of the serde test

Needs follow-up to lookup schema in EventBridge Schema Registry. If the schema says this is never `null`, then yeah this is how it should be done!

Reply:
The aws.ecs@ECSContainerInstanceStateChange schema says nullable, so I remove this function. https://github.com/aws/aws-lambda-go/pull/502/commits/f9ac9df1c93423728f2c29df03f9dd6d4ce73298
ref:
https://eu-west-1.console.aws.amazon.com/events/home?region=eu-west-1#/registries/aws.events/schemas/aws.ecs%40ECSContainerInstanceStateChange

>"stringSetValue": {
            "type": "array",
            "items": {
              "type": "string",
              "nullable": true
            }
          },